minidec/main.rs

```diff
@@ -1,21 +1,28 @@
 extern crate radeco_lib;
 extern crate r2pipe;
+extern crate clap;
 
-use std::path::{Path, PathBuf};
-use std::fs::{self, File};
-use std::io::Write;
+mod cli;
 
 use r2pipe::r2::R2;
-
-use radeco_lib::frontend::containers::RadecoModule;
-use radeco_lib::analysis::sccp;
 use radeco_lib::analysis::cse::CSE;
+use radeco_lib::analysis::sccp;
+use radeco_lib::frontend::containers::RadecoModule;
 use radeco_lib::middle::dce;
 use radeco_lib::middle::ir_writer::IRWriter;
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
 
 fn main() {
+    // Creates cmd args settings then gets the matching params
+    let matches = cli::create_args().get_matches();
+
     let mut dir;
-    let mut r2 = R2::new::<String>(None).expect("Unable to open r2");
+    let mut r2 = R2::new::<String>(Some("/home/ahmed/radare2/radeco-lib/ex-bins/hello-linux-x86_64"
```

Comment:
I don't think this should be hardcoded.

`None` is passed when you want `r2::R2` to use the currently opened file inside r2.
If you want to open a specific file to analyze, you could have it as another argument.